src/utilities.jl

```diff
@@ -317,6 +331,33 @@ printmethod(b, f, m) = String(take!(printmethod(IOBuffer(), b, f, m)))
 get_method_source(m::Method) = Base.uncompressed_ast(m)
 nargs(m::Method) = m.nargs
 
+function isvarargtype(t)
+    @static if VERSION > v"1.7-"
+        t isa Core.TypeofVararg
+    elseif VERSION > v"1.5-"
+        t isa Type && t <: Vararg
+    else
+        # don't special print Vararg
+        # below 1.5
+        false
+    end
+end
+
+function vararg_eltype(t)
+    @static if VERSION > v"1.7-"
+        return t.T
+    elseif VERSION > v"1.5-"
+        if t isa DataType
+            return t.parameters[1]
+        elseif t isa UnionAll
+            return t.body.parameters[1]
+        else
+            error("invalid Vararg type: $t")
```

Comment:
Maybe these should just return Any instead of throwing?

Reply:
yeah make sense for the doc string I guess. I put an error there actually just because I wasn't sure how 1.5- actually handle the `Vararg`.